src/client/csfle/state_machine.rs

```diff
@@ -89,11 +107,21 @@ impl CryptExecutor {
                         Error::internal("db required for NeedMongoMarkings state")
                     })?;
                     let op = RawOutput(RunCommand::new_raw(db.to_string(), command, None, None)?);
-                    let mongocryptd_client = self
-                        .mongocryptd_client
+                    let mongocryptd = self
+                        .mongocryptd
                         .as_ref()
                         .ok_or_else(|| Error::internal("mongocryptd client not found"))?;
-                    let response = mongocryptd_client.execute_operation(op, None).await?;
+                    let mut result = mongocryptd.client.execute_operation(op.clone(), None).await;
+                    if let Err(e) = &result {
+                        if matches!(&*e.kind, ErrorKind::ServerSelection { .. }) {
+                            mongocryptd.respawn().await?;
+                            let new_result = mongocryptd.client.execute_operation(op, None).await;
```

Comment:
so the spec says:
> If the MongoClient fails to connect after spawning, the server selection error is propagated to the user.

is it possible that we successfully connect to the respawned mongocryptd here, but then `execute_operation` fails with a non-server-selection error? if so, should we be propagating that rather than the original server selection error?

Reply:
Good thought, changed to that.  Also reorganized this code block a little to (hopefully) be easier to read.